api/services/TagPresenter.js

```diff
@@ -44,3 +44,40 @@ export const fetchAndPresentForLeftNav = (communityId, userId) =>
         ? merge(c, {new_post_count: find(followed, f => f.name === c.name).new_post_count})
         : c)
     }))
+
+const mostActiveMembers = (community, tag) => {
+  return User.query(q => {
+    q.select(bookshelf.knex.raw('users.name, users.id, users.avatar_url, count(*)'))
+    q.join('post', 'post.user_id', '=', 'users.id')
+    q.join('posts_tags', 'posts_tags.post_id', '=', 'post.id')
+    q.where('tag_id', '=', tag.id)
+    q.groupBy('users.id')
+    q.orderBy('count', 'desc')
+    q.limit(3)
+  })
+  .fetchAll()
+  .then(users => Promise.map(users.models, user => ({
+    id: user.id,
+    name: user.get('name'),
+    avatar_url: user.get('avatar_url'),
+    post_count: user.get('count')
+  })))
+}
+
+export const fetchAndPresentSummary = (community, tag) =>
+  Promise.join(
+    CommunityTag.where({community_id: community.id, tag_id: tag.id})
+    .fetch(),
+    TagFollow.where({community_id: community.id, tag_id: tag.id})
+    .fetchAll(),
+    Search.forPosts({
+      communities: [community.id],
+      tag: tag.id
+    }).fetchAll(),
```

Comment:
@robbiecarlton i missed this in the original review -- using fetchAll here is really expensive, sending a lot of data that you're not going to use since all you care about is the count. It's also buggy because Search.forPosts sets a limit of 20 if you don't provide one explicitly.